src/flask_caching/__init__.py

```diff
@@ -406,6 +411,16 @@ def decorated_function(*args, **kwargs):
                         raise
                     logger.exception("Exception possibly due to cache backend.")
                     return self._call_fn(f, *args, **kwargs)
+                if found and self.app.debug:
+                    logger.info(f"Cache used for key: {cache_key}")
+                if response_hit_indication:
+
+                    def apply_caching(response):
+                        if found:
+                            response.headers["hit_cache"] = found
+                        return response
+
+                    self.app.after_request_funcs[None].append(apply_caching)
```

Comment:
`after_request_funcs` are not cleared at the end of the request and then run for other endpoints that don't even have a cache decorator